### PR TITLE
Chart labels & responsiveness

### PR DESCRIPTION
**Pull request by chelshaw**
This PR makes a few updates to the spider chart results page:

- moves spider chart to a component (in the future we'll have the suggested next steps on the results page)
- adds the labels for each of the dynamics
- updates the chart math a bit so the answers start at the top middle and move clockwise
- makes the chart responsive
- light styling

**Labels with updated styling**
Screenshot 2024-11-06 at 10 25 43 AM

**Small width screen:**
Screenshot 2024-11-07 at 1 05 02 PM

### CODE REVIEW

File: src/lib/components/SpiderChart.svelte [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+<svelte:window bind:innerWidth />
+<div class="outer" style={`padding: ${config.labelD}px;`}>
+	<svg id="chart" width={config.d + 2 * config.p} height={config.d + 2 * config.p}>
+		<g id="ticks">
+			{#each config.ticks as tick}
+				<!-- concentric octogons -->
+				<polygon points={tickToPolygon(tick)} />
+			{/each}
+			{#each radialTickLines() as f}
```
Comment (li-helen) [inline src/lib/components/SpiderChart.svelte:134]: 😮 cool that the return of `radialTickLines` doesn't have to stored in a named var before being used

File: src/routes/results/[results_string]/+page.svelte [hunk 2]
```diff
@@ -1,121 +1,10 @@
...
-		{/each}
-		<path stroke-width="3" stroke="navy" fill="navy" opacity="0.8" d={drawAnswerShape(answers)} />
-		{#each formattedAnswers as ans}
-			<circle cx={ans.xCoord} cy={ans.yCoord} fill="navy" r={config.answerR}></circle>
-			<text x={ans.xCoord - 5} y={ans.yCoord + 3} fill="white">{ans.answer}</text>
-		{/each}
-	</svg>
-</div>
+<main>
+	<SpiderChart answers={data.object} features={Object.keys(data.object)} />
```
Comment (li-helen) [inline src/routes/results/[results_string]/+page.svelte]: nit: could we just pass `data.object`, and derive its keys within the SpiderChart component so we don't need to pass both props?

**Review by li-helen — APPROVED** [review]

**Comment by li-helen** [comment]
looks awesome! 🤩

File: src/lib/components/SpiderChart.svelte [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	<div class="labels-container" style={`width: ${config.labelD}px; height: ${config.labelD}px;`}>
+		{#each labels as label, idx}
+			<div
+				class="dynamic"
+				style={`width: ${config.labelD}px; height: ${config.labelD}px; transform: translate(${label.offsetX}px, ${label.offsetY}px)`}
+			>
+				{#if innerWidth >= BREAKPOINT}
+					{label.text}
+				{:else}
+					Dynamic {idx + 1}
```
Comment (emikjackson) [inline src/lib/components/SpiderChart.svelte:160]: Love having this shortened string for small screens! Not for this PR, but I wonder if a short keyword string would work for each dynamic, or maybe a popup on click to see the full name.

**Review by emikjackson — APPROVED** [review]
amazing lgtm!

**Pull request merged**